framework/branch-key-store.md

```diff
@@ -418,97 +416,205 @@ This operation MUST return the constructed [branch key materials](./structures.m
+and the version string is this value.
+If the `type` attribute start with `"branch:version:"` then the version string MUST be equal to this value.
+
+To construct [beacon key materials](./structures.md#beacon-key-materials) from authenticated encryption context as follows:
+
+- [Branch Key](./structures.md#branch-key) MUST be the [decrypted branch key material](#aws-kms-branch-key-decryption)
+- [Branch Key Id](./structures.md#branch-key-id) MUST be the `branch-key-id`
+- [Branch Key Version](./structures.md#branch-key-version)
+  The version string MUST start with `branch:version:`.
+  The remaining string encoded as UTF8 bytes MUST be the Branch Key version.
+- [Encryption Context](./structures.md#encryption-context-3) MUST be constructed by
+  [Custom Encryption Context From Authenticated Encryption Context](#custom-encryption-context-from-authenticated-encryption-context)
+
+### Custom Encryption Context From Authenticated Encryption Context
+
+The custom encryption context is stored as map of UTF8 Encoded bytes.
+
+For every key in the [encryption context](./structures.md#encryption-context-3)
+the string `aws-crypto-ec:` + the UTF8 decode of this key
+MUST exist as a key in the authenticated encryption context.
+Also, the value in the [encryption context](./structures.md#encryption-context-3) for this key
+MUST equal the value in the authenticated encryption context
+for the constructed key.
+
+### Example
+
+Given the simplified [branch key material](./structures.md#branch-key-materials) structure
+
+```dafny
+BranchKeyMaterials(
+  branchKey := [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  branchKeyId := "bbb9baf1-03e6-4716-a586-6bf29995314b",
+  branchKeyVersion := "83eec007-5659-4554-bf11-699b90f41ac6",
+  encryptionContext := [ "department" := "admin"]
+)
+```
+
+There would be three items in the keystore table.
+The DECRYPT_ONLY version, the ACTIVE version, and a beacon key.
+
+The DECRYPT_ONLY simplified JavaScript JSON format would look like this
+
+```
+{
+  "branch-key-id" : "bbb9baf1-03e6-4716-a586-6bf29995314b",
+  "type" : "branch:version:83eec007-5659-4554-bf11-699b90f41ac6",
+  "enc" : "NnYwxJ/oiQCLnqRh/IcrCR2mmOnO4SAVLw2pspKJKd6rpa0H8z/4hGpGxcWozdb7VByebDFWb0VTWxaOUA8=",
+  "kms-arn" : "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+  "create-time" : "2023-06-03T19:03:29.358Z",
+  "hierarchy-version" : "1",
+  "aws-crypto-ec:department" : "admin",
+}
+```
+
+The ACTIVE simplified JavaScript JSON format would look like this
+
+```
+{
+  "branch-key-id" : "bbb9baf1-03e6-4716-a586-6bf29995314b",
+  "type" : ""branch:ACTIVE",
+  "enc" : "BiXHTm0j27+jsgJZ7yCnvI6yvjFyStMsHiC8fnR9KzKjwwhi0gB+5CZTfXFC2ufmBtCYX/sLvKsFnEITR+k=",
+  "kms-arn" : "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+  "create-time" : "2023-06-03T19:03:29.358Z",
+  "hierarchy-version" : "1",
+  "aws-crypto-ec:department" : "admin",
+}
```

Comment:
missing the version field on the active record
```suggestion
{
  "branch-key-id" : "bbb9baf1-03e6-4716-a586-6bf29995314b",
  "type" : "branch:ACTIVE",
  "version": "83eec007-5659-4554-bf11-699b90f41ac6",
  "enc" : "BiXHTm0j27+jsgJZ7yCnvI6yvjFyStMsHiC8fnR9KzKjwwhi0gB+5CZTfXFC2ufmBtCYX/sLvKsFnEITR+k=",
  "kms-arn" : "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab",
  "create-time" : "2023-06-03T19:03:29.358Z",
  "hierarchy-version" : "1",
  "aws-crypto-ec:department" : "admin",
}
```

Reply:
```suggestion
{
  "branch-key-id" : "bbb9baf1-03e6-4716-a586-6bf29995314b",
  "type" : ""branch:ACTIVE",
  "version": "branch:version:83eec007-5659-4554-bf11-699b90f41ac6"
  "enc" : "BiXHTm0j27+jsgJZ7yCnvI6yvjFyStMsHiC8fnR9KzKjwwhi0gB+5CZTfXFC2ufmBtCYX/sLvKsFnEITR+k=",
  "kms-arn" : "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab",
  "create-time" : "2023-06-03T19:03:29.358Z",
  "hierarchy-version" : "1",
  "aws-crypto-ec:department" : "admin",
}
```